src/java/com/zimbra/ssdb/SSDBValueEncoder.java

```diff
@@ -15,12 +17,27 @@ public String encodeValue(EphemeralInput input, EphemeralLocation target) {
         if(input == null || input.getValue() == null) {
             return null;
         }
+
         Long expires = input.getExpiration();
         String value = input.getValue().toString();
+        String encoded;
+
         if (expires != null && expires > 0L) {
-            return String.format("%s|%s", value, String.valueOf(expires));
+            encoded = String.format("%s|%s", value, String.valueOf(expires));
         } else {
-            return String.format("%s|", value);
+            encoded = String.format("%s|", value);
+        }
+
+        if (!LC.ssdb_zimbrax_compat.booleanValue()) {
+            return encoded;
+        }
+
+        if (encoded.startsWith("{") || encoded.startsWith("[")) {
+            // Already a JSON object
```

Comment:
For these purposes, I think so.  If it doesn't start with those characters, it can be encoded as a string and parsed properly by ZOK.  We don't currently store any raw numbers in ephemeral data.